src/docfx/build/redirection/RedirectionMap.cs

```diff
@@ -93,9 +94,14 @@ void AddRedirections(Dictionary<string, SourceInfo<string>> items, bool redirect
                                 errors.Add(Errors.RedirectionUrlInvalid(redirectUrl));
                                 continue;
                         }
+
+                        if (redirectUrl.Value.EndsWith("index"))
```

Comment:
this check also matches `reindex`, do it in `Document` after absolute URL is combined, then check against `/index`